src/random_unitary.jl

```diff
@@ -0,0 +1,69 @@
+# Version of `sign` that returns one
+# if `x == 0`.
+function nonzero_sign(x)
+  iszero(x) && return one(x)
+  return sign(x)
+end
+
+using LinearAlgebra: LinearAlgebra, Diagonal, diag
+function qr_positive(M::AbstractMatrix)
+  Q, R = LinearAlgebra.qr(M)
+  Q′ = typeof(R)(Q)
+  signs = nonzero_sign.(diag(R))
+  Q′ = Q′ * Diagonal(signs)
+  R = Diagonal(conj.(signs)) * R
+  return Q′, R
+end
+
+using Random: Random, AbstractRNG
+
+dual(x) = x
```

Comment:
Do we need this here? It seems a bit strange to require a definition of something that is related to symmetries.

This is also somewhat related to the convention choice of GradedUnitRanges btw: if you make `dual` signify whether or not the vector space is dual, and not whether or not the arrow is incoming or outgoing, defining a unitary would be `zeros(elt, ax, ax)` instead of the `zeros(elt, dual(ax), ax)`

Reply:
That's definitely an interesting point that it ties into the convention we use for how we specify the duality of the domain/codomain.

Reply:
Related to your suggestion of making a special overload of `zeros`, we could define constructors for maps or operators, like:
```julia
zero_op(elt, ax) = zeros(elt, ax
```

Reply:
Oh wait, I got confused here, `zeros` is only used in `TensorAlgebraGradedUnitRangesExt`. It's really the `splitdims` function that needs to know in some way that the codomain is the dual of the domain.